### ISSUE TITLE
dp freeze not work with horovod-parallel trained model with deepmd-kit-2.0.1-gpu

### ISSUE THREAD

**Issue by metaliuyong** [deepmodeling/deepmd-kit#1336]
The train process is normal, but when I freeze the model it comes problems as follow:
2021-12-05 09:03:43.553693: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
[0] 2021-12-05 09:03:49.426510: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
[0] WARNING:tensorflow:From /home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
[0] Instructions for updating:
[0] non-resource variables are not supported in the long term
[0] Traceback (most recent call last):
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/bin/dp", line 10, in <module>
[0]     sys.exit(main())
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/deepmd/entrypoints/main.py", line 439, in main
[0]     freeze(**dict_args)
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/deepmd/entrypoints/freeze.py", line 162, in freeze
[0]     saver = tf.train.import_meta_graph(
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/training/saver.py", line 1465, in import_meta_graph
[0]     return _import_meta_graph_with_return_elements(meta_graph_or_file,
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/training/saver.py", line 1486, in _import_meta_graph_with_return_elements
[0]     meta_graph.import_scoped_meta_graph_with_return_elements(
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/framework/meta_graph.py", line 801, in import_scoped_meta_graph_with_return_elements
[0]     imported_return_elements = importer.import_graph_def(
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/util/deprecation.py", line 535, in new_func
[0]     return func(*args, **kwargs)
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/framework/importer.py", line 400, in import_graph_def
[0]     return _import_graph_def_internal(
[0]   File "/home/jmxue_pkuhpc/lustre3/anaconda3/envs/deepmd-kit-2.0.1-gpu/lib/python3.9/site-packages/tensorflow/python/framework/importer.py", line 496, in _import_graph_def_internal
[0]     results = c_api.TF_GraphImportGraphDefWithResults(
[0] tensorflow.python.framework.errors_impl.NotFoundError: Op type not registered 'HorovodAllreduce' in binary running on login12. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.

**Comment by njzjz** [comment]
Duplicate of #1193.
